src/UNL/VisitorChat/Util.php

```diff
@@ -0,0 +1,30 @@
+<?php
+namespace UNL\VisitorChat;
+
+class Util
+{
+    public static function isAllowedDomain($url, array $domains)
+    {
+        if (!$parts = parse_url($url)) {
+            //it isn't a valid URL, don't allow
+            return false;
+        }
+        
+        if (!isset($parts['host'])) {
+            //it isn't a valid host, don't allow
+            return false;
+        }
+
+        $regex = "";
+        foreach ($domains as $domain) {
+            $regex .= preg_quote($domain, ".-/") . "|";
+        }
+
+        $regex = trim($regex, "|");
+        if (preg_match("/" . $regex . "$/", $parts['host'])) {
```

Comment:
You might also want to add a constrained for better detecting sub-domains. What happens when someone uses the domain `iamevilprefixtest.com`